lib/tasks/analytics.rake

```diff
@@ -0,0 +1,102 @@
+      # we use COALESCE function to add null value as currently we find on occassions google data studio struggles with empty values.
+
+      # REMOVE THESE: once we fix the data dashboard to match dynamic keys we can remove these from sql
+      # COALESCE(module_time_to_completion->>'child-development-and-the-eyfs', 'null') AS module_1_time,
+      # COALESCE(module_time_to_completion->>'brain-development-and-how-children-learn', 'null') AS module_2_time,
+      # COALESCE(module_time_to_completion->>'personal-social-and-emotional-development', 'null') AS module_3_time,
+      # COALESCE(module_time_to_completion->>'module-4', 'null') AS module_4_time,
+
+      sql = 'SELECT id, module_time_to_completion as json_column, (SELECT COUNT(*) FROM jsonb_object_keys(module_time_to_completion)) nbr_keys FROM public.users order by nbr_keys desc limit 1'
+      user_json = ActiveRecord::Base.connection.execute(sql)
+      users_all = User.select(" id,
+                                TO_CHAR(created_at, 'YYYY-MM-DD') AS registered_at,
+                                COALESCE(NULLIF(setting_type, 'other'), COALESCE(setting_type_other, 'null')) AS user_setting,
+                                COALESCE(module_time_to_completion->>'child-development-and-the-eyfs', 'null') AS module_1_time,
+                                COALESCE(module_time_to_completion->>'brain-development-and-how-children-learn', 'null') AS module_2_time,
+                                COALESCE(module_time_to_completion->>'personal-social-and-emotional-development', 'null') AS module_3_time,
+                                COALESCE(module_time_to_completion->>'module-4', 'null') AS module_4_time,
+                                #{AnalyticsBuild.build_json_sql('module_time_to_completion', JSON.parse(user_json.first['json_column']))}
+                                module_time_to_completion").all
+
+      users = AnalyticsBuild.new(bucket_name: ENV['GCS_BUCKET_NAME'],
+                                 folder_path: 'userdata',
+                                 result_set: users_all, file_name: 'users')
+
+      users.create! if Rails.env.test?
+      users.delete_files if Rails.env.production? || Rails.env.development?
+      users.upload if Rails.env.production? || Rails.env.development?
+      Rake::Task['db:analytics:users'].reenable # need to reset rake task before it can be run again.
+    end
+
+    desc 'ahoy_events table'
+    task ahoy_events: :environment do
+      sql = 'SELECT id, properties as json_column, (SELECT COUNT(*) FROM jsonb_object_keys(properties)) nbr_keys FROM public.ahoy_events order by nbr_keys desc limit 1'
+      event_json = ActiveRecord::Base.connection.execute(sql)
+      events_names_list = %w[summative_assessment_start confidence_check_complete confidence_check_start module_complete user_note_updated user_note_created user_name_change summative_assessment_complete user_registration questionnaire_answer]
```

Comment:
Is there a benefit to doing it this way rather than just excluding the generic page view event `module_content_page`?
For example, doing it this way means any new named events need to be explicitly added otherwise they'll not be fed to the board. Your list is missing `module_start`.

Reply:
No, page type and name are properties.